### PR TITLE
New module antennadedispersion

### PR DESCRIPTION
**Pull request by cg-laser**
Adds a new module that dedisperses antennas.

Also fixed a typo in the detector module in the function `get_antanna_orientation`

### CODE REVIEW

File: NuRadioReco/modules/channelAntennaDedispersion.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    def _get_response(self, det, station_id, channel_id, ff):
+        antenna_name = det.get_antenna_model(station_id, channel_id)
+        antenna = self._provider.load_antenna_pattern(antenna_name)
+        zen_ori, az_ori, zen_rot, az_rot = det.get_antenna_orientation(station_id, channel_id)
+        zen, az = None, None
+        if("LPDA" in antenna_name):
+            zen = zen_ori
+            az = az_ori
+        elif("bicone" in antenna_name or "dipole" in antenna_name):
+            zen = 90 * units.deg + zen_ori
```
Comment (christophwelling) [inline NuRadioReco/modules/channelAntennaDedispersion.py]: Is it documented somewhere that LPDA and bicone have their orientation defined differently?

File: NuRadioReco/modules/channelAntennaDedispersion.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    def run(self, evt, station, det):
+        """
+        parameters
+        ----------
+        """
+        for channel in station.iter_channels():
+            ff = channel.get_frequencies()
+            response = self._get_response(det, station.get_id(), channel.get_id(), tuple(ff))
+
+#             trace = channel.get_trace()
```
Comment (christophwelling) [inline NuRadioReco/modules/channelAntennaDedispersion.py]: Please remove this if it is not needed

File: NuRadioReco/modules/channelAntennaDedispersion.py (hunk 1 above)
Comment (anelles) [reply NuRadioReco/modules/channelAntennaDedispersion.py]: We should really make this consistently, or issue some warnings somewhere. I was not aware that this works like this.

File: NuRadioReco/modules/channelAntennaDedispersion.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        polarization = "phi"
+        if(np.sum(np.abs(VEL['theta'])) > np.sum(np.abs(VEL['phi']))):
+            polarization = "theta"
+        response = np.exp(1j * np.angle(VEL[polarization]))
+#         fig, ax = plt.subplots(1, 1)
+#         ax.plot(ff / units.MHz, np.abs(VEL['theta']), label='theta')
+#         ax.plot(ff / units.MHz, np.abs(VEL['phi']), label='phi')
+#         ax.set_title(f"{antenna_name} - {polarization}")
+#         ax.legend()
+#         plt.show()
```
Comment (anelles) [inline NuRadioReco/modules/channelAntennaDedispersion.py]: Clean up, please.

**Comment by christophwelling** [comment]
It's a typo fix. antAnna -> antEnna

File: NuRadioReco/modules/channelAntennaDedispersion.py (hunk 1 above)
Comment (cg-laser) [reply NuRadioReco/modules/channelAntennaDedispersion.py]: The orientation is not defined differently. The first vector is the orientation/direction of the antenna. But the sensitive direction for LPDAs and dipoles is different (by exactly 90 degrees). The LPDA is most sensitive when the signal comes from the boresight direction. A dipole is most sensitive when the signal comes in perpendicular to the antenna orientation (thus +90 deg + antenna orientation zenith angle).

**Comment by cg-laser** [comment]
oki doki, I cleaned up the module. 
I also added a check to the efieldToVoltage converter to check if a station has any efields.

**Comment by anelles** [comment]
Thanks, can you update the docstring in get_antenna_orientation ? It now talks about connectors. If we add here, what we mean for dipole and LPDA, I think this would alleviate some confusion.

**Comment by cg-laser** [comment]
> Thanks, can you update the docstring in get_antenna_orientation ? It now talks about connectors. If we add here, what we mean for dipole and LPDA, I think this would alleviate some confusion.

@anelles done

**Pull request merged**